src/machine/image-dbus.c

```diff
@@ -56,11 +61,37 @@ int bus_image_method_remove(
         if (r == 0)
                 return 1; /* Will call us back */
 
-        r = image_remove(image);
-        if (r < 0)
+        if (pipe2(errno_pipe_fd, O_CLOEXEC|O_NONBLOCK) < 0)
+                return sd_bus_error_set_errnof(error, errno, "Failed to create pipe: %m");
+
```

Comment:
i can see the usefulness of this, but let's maybe revisit this when this comes up again. I think as long as we get away without this in a couple of times things are OK...